aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -935,6 +951,81 @@ std::tuple<Tensor&, Tensor&> symeig_out(Tensor& vals, Tensor& vecs, const Tensor
   return std::tuple<Tensor&, Tensor&>(vals, vecs);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ eig ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+template <typename scalar_t>
+static void apply_eig(Tensor& vals_, Tensor& vecs_, const Tensor& self, bool eigenvectors) {
+#ifndef USE_LAPACK
+  AT_ERROR("symeig: LAPACK library not found in compilation");
+#else
+  int n = self.size(0);
+  Tensor wi = at::empty({n}, self.dtype());
+  Tensor wr = at::empty({n}, self.dtype());
+  char jobvr = eigenvectors ? 'V' : 'N';
+  int lda = n;
+  int info;
+  if (n > 0) {
+    scalar_t wkopt;
+    scalar_t* vecs_data = eigenvectors ? vecs_.data<scalar_t>() : nullptr;
+    int ldvr = eigenvectors ? n : 1;
+
+    lapackEig<scalar_t>('N', jobvr, n, self.data<scalar_t>(), lda, wr.data<scalar_t>(), wi.data<scalar_t>(),
+      nullptr, 1, vecs_data, ldvr, &wkopt, -1, &info);
+
+    int lwork = (int)wkopt;
+    Tensor work = at::empty({lwork}, self.dtype());
+
+    lapackEig<scalar_t>('N', jobvr, n, self.data<scalar_t>(), lda, wr.data<scalar_t>(), wi.data<scalar_t>(),
+      nullptr, 1, vecs_data, ldvr, work.data<scalar_t>(), lwork, &info);
+  }
+
+  Tensor wr_ = at::slice(vals_, 1, 0, 1).squeeze(1);
+  Tensor wi_ = at::slice(vals_, 1, 1, 2).squeeze(1);
+  wr_.copy_(wr);
+  wi_.copy_(wi);
+#endif
+}
+
+static std::tuple<Tensor, Tensor> _eig_helper_cpu(const Tensor& self, bool eigenvectors) {
+  TORCH_CHECK(self.dim() == 2, "A should be 2 dimensional");
+  TORCH_CHECK(self.size(0) == self.size(1), "A should be square");
+  TORCH_CHECK(self.isfinite().all().item<bool>(), "A should not contain infs or NaNs");
+  int n = self.size(0);
+  Tensor vecs_;
+  if (eigenvectors) {
+    vecs_ = at::empty({n, n}, self.dtype()).transpose(0, 1);
+  } else {
+    vecs_ = at::empty({0}, self.dtype());
+  }
+  Tensor vals_ = at::empty({n, 2}, self.dtype());
+  Tensor self_ = cloneBatchedColumnMajor(self);
+
+  AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), "eig_cpu", [&]{
+    apply_eig<scalar_t>(vals_, vecs_, self_, eigenvectors);
+  });
+  return std::tuple<Tensor&, Tensor&>(vals_, vecs_);
+}
+
+std::tuple<Tensor&, Tensor&> eig_out_cpu(Tensor& vals, Tensor& vecs, const Tensor& self, bool eigenvectors) {
+  TORCH_CHECK(self.dtype() == vals.dtype(), "dtype of eigenvalues tensor must match input");
+  TORCH_CHECK(!eigenvectors || (self.dtype() == vecs.dtype()), "dtype of eigenvectors tensor must match input");
+  Tensor vals_tmp, vecs_tmp;
+  std::tie(vals_tmp, vecs_tmp) = _eig_helper_cpu(self, eigenvectors);
+  vals.resize_as_(vals_tmp).copy_(vals_tmp);
+  if (eigenvectors) {
+    // Need vector output to be in transposed-contiguous format
+    vecs.resize_as_(vecs_tmp.transpose(0, 1), MemoryFormat::Contiguous).transpose_(0, 1);
```

Comment:
This is still not right, if vecs is correct size you should not be changing its strides, you have to use it directly (even if its strides are different from returned vecs_tmp). 
If vecs was the wrong size to start with, then it's ok to make its strides match vecs_tmp. Also, can you please use at::native::resize here to make sure that warning is thrown if vecs size is wrong?

Reply:
I'm not sure I understand. The old TH code was using `THTensor_(newTransposedContiguous)`, then setting the data pointer of the output eigenvectors tensor to the new transposed-contiguous tensor. I thought this code would give the same result. But now I'm not sure what is the correct way to do it.

Reply:
It is possible that old code was doing it wrong, but current behavior that we are trying to enforce is to not change the strides of the `out` tensors as long as they are of the correct size.